plugins/aggregators/valuecounter/valuecounter.go

```diff
@@ -80,7 +95,16 @@ func (vc *ValueCounter) Push(acc telegraf.Accumulator) {
 		fields := map[string]interface{}{}
 
 		for field, count := range agg.fieldCount {
-			fields[field] = count
+			var matched = 0
+			for _, predicate := range vc.Predicates {
+				if predicate.predicateFunc(count, predicate.Value) {
+					matched++
+				}
+			}
+
+			if matched == len(vc.Predicates) {
+				fields[field] = count
+			}
```

Comment:
the above block can be simplified by removing `matched`, you could put a `label` for the field loop such as "FieldCount:" and then you could leave the assignment at the end without an if statement wrapping it and just put a `
if !predicate.predicateFunc(count, predicate.Value) {
	continue FieldCount
}
` on line 100. That way time isn't spent doing checks even after one failed.

Reply:
Great suggestion.   Hadn't even crossed my mind.
cheers
/dom

Reply:
!retry-failed

Reply:
Apologies I forgot to add a note here.  I made the suggested changes.  Let me know if this wasn't matching what you had in mind; happy to update if needs.

cheers
/dom